src/peer_manager.rs

```diff
@@ -254,6 +254,7 @@ impl Peer {
     fn is_joining_node(&self) -> bool {
         match self.state {
             PeerState::JoiningNode => true,
+            PeerState::Node(val) => val,
```

Comment:
This feels wrong somehow. A peer with `PeerState::Node` is a full node, not a joining node. The fact that this function can return `true` for a full node is surprising and confusing. I'd suggest a different approach: revert this function as it was and instead add a new function: `was_joining_node` or `is_joined_node` or whatever, which returns true only if `self.state == PeerState::Node(true)`.

Reply:
Most use of this need bot functions, to work together. I've renamed to is_or_was_joining_node.